docs(reactnative/setup): update with correct instructions (#5026)
aws-amplify/docs: src/fragments/start/getting-started/reactnative/auth.mdx

```bash
npm i @aws-amplify/ui-react-native aws-amplify react-native-safe-area-context amazon-cognito-identity-js @react-native-community/netinfo @react-native-async-storage/async-storage react-native-get-random-values react-native-url-polyfill
```

### Add the Amplify UI Authenticator component

<BlockSwitcher>
<Block name="Expo CLI">

Open **App.js** and make the following changes:

</Block>
<Block name="React Native CLI">

Open **App.ts** and make the following changes:

</Block>
</BlockSwitcher>

1. Import the `withAuthenticator` component and `useAuthenticator` hook:

```javascript

import { Authenticator, useAuthenticator } from '@aws-amplify/ui-react-native';

````

2. Pass `{signOut, user}` to the any component where you need it (e.g. `SignOutButton`):

```javascript

const SignOutButton = () => {
  const { user, signOut } = useAuthenticator((context) => [context.user]);
  return (
    <Pressable onPress={signOut} style={styles.buttonContainer}>
      <Text style={styles.buttonText}>Hello, {user.username}! Click here to sign out!</Text>
    </Pressable>
  )
};

```

3. Add `SignOutButton` component to  your `App` component:

```javascript

// ...
return (

>     <View style={styles.container}>
>       <StatusBar hidden/>
>       <SignOutButton />

Reviewer: ```suggestion
  <View style={styles.container}>
    <SignOutButton />
```

Fixed spacing and also recommend removing the StatusBar component as it is not imported into the project and not needed

```javascript
import React, { useEffect, useState } from 'react'
import { StyleSheet, Text, View, TextInput, Pressable } from 'react-native';
import { API, graphqlOperation } from 'aws-amplify'
import { createTodo } from './src/graphql/mutations'
import { listTodos } from './src/graphql/queries'
import { withAuthenticator, useAuthenticator } from '@aws-amplify/ui-react-native';

const SignOutButton = () => {
  const { user, signOut } = useAuthenticator((context) => [context.user]);
  return (
    <Pressable onPress={signOut} style={styles.buttonContainer}>
      <Text style={styles.buttonText}>Hello, {user.username}! Click here to sign out!</Text>
    </Pressable>
  )
};

const initialState = { name: '', description: '' }

const App = () => {
  const [formState, setFormState] = useState(initialState)
  const [todos, setTodos] = useState([])

  useEffect(() => {
    fetchTodos()
  }, [])

  function setInput(key, value) {
    setFormState({ ...formState, [key]: value })
  }

  async function fetchTodos() {
    try {
      const todoData = await API.graphql(graphqlOperation(listTodos))
      const todos = todoData.data.listTodos.items
      setTodos(todos)

>     } catch (err) { console.log('error fetching todos') }

Reviewer: ```suggestion
    } catch (err) { 
      console.log('error fetching todos') 
    }
```

        style={styles.input}
        value={formState.description}
        placeholder="Description"
      />
      <Pressable onPress={addTodo} style={styles.buttonContainer}>
        <Text style={styles.buttonText}>Create todo</Text>
      </Pressable>
      {
        todos.map((todo, index) => (
          <View key={todo.id ? todo.id : index} style={styles.todo}>
            <Text style={styles.todoName}>{todo.name}</Text>
            <Text style={styles.todoDescription}>{todo.description}</Text>
          </View>
        ))
      }
    </View>
  );
}

export default withAuthenticator(App)

const styles = StyleSheet.create({
  container: { width: 400, flex: 1, padding: 20, alignSelf: 'center' },
  todo: {  marginBottom: 15 },
  input: { backgroundColor: '#ddd', marginBottom: 10, padding: 8, fontSize: 18 },
  todoName: { fontSize: 20, fontWeight: 'bold' },
  todoDescription: { marginBottom: 0 },
  buttonContainer: { alignSelf: 'center', backgroundColor: 'black',  paddingHorizontal: 8 },
  buttonText: { color: 'white', padding: 16, fontSize: 18 },
});
```

Using Amplify UI connected components together makes it easier to manage styling across your entire app.

> In this example, you used the Amplify React UI library and the `withAuthenticator` component to quickly get up and running with a real-world authentication flow.

Reviewer: ```suggestion
In this example, you used the Amplify React UI library and the `withAuthenticator` Higher-Order Component to quickly get up and running with a real-world authentication flow.
```

You can also [customize this component](https://ui.docs.amplify.aws/react-native/connected-components/authenticator/customization) to add or remove fields, update styling, or other configurations. You can even [override function calls](https://ui.docs.amplify.aws/react-native/connected-components/authenticator/customization#override-function-calls) if needed.

> In addition to the `withAuthenticator`, you can build [custom authentication flows Amplify Libraries for JS](/lib/auth/getting-started). Amplify's `Auth` class has over 30 methods including `signUp`, `signIn`, `forgotPassword`, and `signOut` that allow you full control over all aspects of the user authentication flow. Check out the complete API [here](https://aws-amplify.github.io/amplify-js/api/classes/authclass.html).

Reviewer: ```suggestion
In addition to `withAuthenticator`, you can build [custom authentication flows with the Amplify Library for JS](/lib/auth/getting-started). Amplify's `Auth` class has over 30 methods including `signUp`, `signIn`, `forgotPassword`, and `signOut` that allow you full control over all aspects of the user authentication flow. Check out the complete API [here](https://aws-amplify.github.io/amplify-js/api/classes/authclass.html).
```
